lib/vector-core/src/event/metadata.rs

```diff
@@ -70,6 +73,16 @@ impl EventMetadata {
         &mut self.secrets
     }
 
+    /// Returns a mutable reference to the metadata source.
+    pub fn source(&self) -> &Option<Arc<OutputId>> {
+        &self.source
+    }
```

Comment:
The `Arc` in the return does not appear to be used. Could this just return a reference?
```suggestion
    pub fn source(&self) -> Option<&OutputId> {
        self.source.as_deref()
    }
```